crates/sui-graphql-rpc/src/zklogin_verify_signature.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::data::Db;
+use crate::error::Error;
+use crate::types::base64::Base64;
+use crate::types::chain_identifier::ChainIdentifier;
+use crate::types::dynamic_field::{DynamicField, DynamicFieldName};
+use crate::types::epoch::Epoch;
+use crate::types::sui_address::SuiAddress;
+use crate::types::type_filter::ExactTypeFilter;
+use async_graphql::*;
+use fastcrypto::encoding::Encoding;
+use fastcrypto::{encoding::Base64 as FastcryptoBase64, traits::ToFromBytes};
+use fastcrypto_zkp::bn254::zk_login_api::ZkLoginEnv;
+use im::hashmap::HashMap as ImHashMap;
+use shared_crypto::intent::{
+    AppId, Intent, IntentMessage, IntentScope, IntentVersion, PersonalMessage,
+};
+use std::str::FromStr;
+use sui_types::authenticator_state::{ActiveJwk, AuthenticatorStateInner};
+use sui_types::digests::{get_mainnet_chain_identifier, get_testnet_chain_identifier};
+use sui_types::dynamic_field::{DynamicFieldType, Field};
+use sui_types::signature::GenericSignature;
+use sui_types::signature::{AuthenticatorTrait, VerifyParams};
+use sui_types::transaction::TransactionData;
+use tracing::warn;
+
+pub(crate) struct ZkLoginVerify;
+
+/// Verifies a zkLogin signature based on the provided transaction or personal message.
+impl ZkLoginVerify {
+    /// Verify a zkLogin signature based on the provided transaction or personal message
+    /// based on current epoch, chain id, and latest JWKs fetched on-chain.
+    ///
+    /// If the signature is valid, the function returns a `ZkLoginVerifyResult` with no
+    /// errors. If the signature is invalid, the function returns a `ZkLoginVerifyResult`
+    /// with a list of errors.
+    ///
+    /// - `bytes` is either the personal message in raw bytes or transaction data bytes in
+    ///    BCS-encoded and then Base64-encoded.
+    /// - `signature` is a serialized zkLogin signature that is Base64-encoded.
+    /// - `intent_scope` is a u64 representing the intent scope of bytes.
+    /// - `author` is the address of the signer of the transaction or personal msg.
+    pub(crate) async fn verify_zklogin_signature(
+        db: &Db,
+        bytes: String,
+        signature: String,
+        intent_scope: u64,
```

Comment:
thought/ suggestion: it seems like we have only two intent scopes, transactiondata or personal message, maybe we can represent this on the schema as an enum instead of a u64?

Reply:
see my other comment, does enum support integer? i would love to use the enum and make it with two options 0 or 3.

Reply:
Resolving this thread and I'll follow up in the other one